go/vt/vtgate/schema/tracker.go

```diff
@@ -134,20 +134,23 @@ func (t *Tracker) loadViews(conn queryservice.QueryService, target *querypb.Targ
 		return nil
 	}
 
-	fvRes, err := conn.GetSchema(t.ctx, target, querypb.SchemaTableType_VIEWS, nil)
-	if err != nil {
-		return err
-	}
-
 	t.mu.Lock()
 	defer t.mu.Unlock()
 	// We must clear out any previous view definition before loading it here as this is called
-	// whenever a shard's primary tablet starts and sends the initial signal. Without
-	// clearing out the previous view definition removes any dropped views.
+	// whenever a shard's primary tablet starts and sends the initial signal.
+	// This is needed clear out any stale view definitions.
 	t.clearKeyspaceViews(target.Keyspace)
-	t.updateViews(target.Keyspace, fvRes)
-	log.Infof("finished loading views for keyspace %s. Found %d views", target.Keyspace, len(fvRes))
 
+	var noOfViews int
+	err := conn.GetSchema(t.ctx, target, querypb.SchemaTableType_VIEWS, nil, func(schemaRes *querypb.GetSchemaResponse) error {
+		t.updateViews(target.Keyspace, schemaRes.TableDefinition)
+		noOfViews += len(schemaRes.TableDefinition)
```

Comment:
this is an extreme nitpick (and especially if this was existing code please ignore it) -- to me, "no." only works as an abbreviation with the period included (which you obviously can't have here), how about `numViews` instead?